src/vs/workbench/common/editor/editorGroup.ts

```diff
@@ -663,6 +664,14 @@ export class EditorGroup extends Disposable {
 					&& !this.configurationService.getValue<boolean>('sql.promptToSaveGeneratedFiles')) {
 					return;
 				}
+				// Do not add generated files from Temp if file is not dirty
+				if (e instanceof FileEditorInput && !e.isDirty()) {
+					let filePath = e.getResource() ? e.getResource().fsPath : undefined;
+					let tempPath = process.env[process.platform === 'win32' ? 'TEMP' : 'TMPDIR'];
```

Comment:
are you pretty confident that this condition will only target the generated files?  i.e., there won't be other files opened from temp that might get included by this condition?

Reply:
Actually I can not be sure of that - I can make the path more restrictive though. Will do that

Reply:
sounds good.  I don't think it's a huge issue either way, since the impact is that temp files won't be reopened on next startup, which isn't a huge deal IMO, but it would be good to scope as much as reasonable to further limit any impact.

Reply:
updated to be mssql_definition folder - so its less possible for any other file to be present